### ISSUE TITLE
Chapter 12: 01_boosting_baseline.ipynb

### ISSUE THREAD

**Issue by JeremyWhittaker** [stefan-jansen/machine-learning-for-trading#120]
I was able to run almost everything in this notebook. Then I got to this section `cat_gpu_result = stack_results(cat_gpu_cv_result)
cat_gpu_result.groupby(['Metric', 'Dataset']).Value.mean().unstack()`

It returns all NaN
![image](https://user-images.githubusercontent.com/13750325/110554684-849c1b80-80f8-11eb-9c9f-d6118e555aab.png)

Same with the next line of code `plot_result(cat_gpu_result, model='CatBoost', fname=f'figures/{algo}_cv_result')`
![image](https://user-images.githubusercontent.com/13750325/110554731-9aa9dc00-80f8-11eb-8460-d395ad03ea55.png)

I'm trying to figure out what I did wrong here.

**Comment by stefan-jansen** [comment]
Do you have a GPU and install CatBoost to use it as described in their [docs](https://catboost.ai/docs/features/training-on-gpu.html)?

**Comment by N2Network** [comment]
I'm able to get all the other code above it to run which I think is based on Catboost. I so I think it's working. I'm using a Docker image so I think most of the pre-requisites have been installed. I'm running a GTX950 with the correct driver version. I read through the Catboost docs and I'm not seeing anything that needs to be configured for my specific graphics card. Perhaps I'm missing something? It seems like it's on the Cross-Validate code section is not working. 

I ran the Catboost test code and it seems to work:
```
import numpy 
from catboost import CatBoostRegressor

dataset = numpy.array([[1,4,5,6],[4,5,6,7],[30,40,50,60],[20,15,85,60]])
train_labels = [1.2,3.4,9.5,24.5]
model = CatBoostRegressor(learning_rate=1, depth=6, loss_function='RMSE')
fit_model = model.fit(dataset, train_labels)

print(fit_model.get_params())
```

```
999:	learn: 0.0000000	total: 1.93s	remaining: 0us
{'learning_rate': 1, 'depth': 6, 'loss_function': 'RMSE'}
```

Example: `plot_result(cat_result, model='CatBoost', fname=f'figures/{algo}_cv_result')`
![image](https://user-images.githubusercontent.com/13750325/110668412-84e3f780-8188-11eb-860d-abee52205d05.png)

**Comment by stefan-jansen** [comment]
I'm sorry, I wasn't very clear. Are you using Docker? Unfortunately, Catboost won't be able to access your GPU via Docker; I've added a warning in the notebook. The code works fine on a local host with GPU, though,  just re-tested.

**Comment by JeremyWhittaker** [comment]
> I'm sorry, I wasn't very clear. Are you using Docker? Unfortunately, Catboost won't be able to access your GPU via Docker; I've added a warning in the notebook. The code works fine on a local host with GPU, though, just re-tested.

I am using Docker. I wasn't aware the VM couldn't pass the GPU. Thanks for the follow up.

**Comment by JeremyWhittaker** [comment]
Alternative fix:
I ended up modifying this code to not use the GPU and it appears to have worked.

# cat_gpu_result = stack_results(cat_gpu_cv_result)
#no access to gpu modified code
cat_gpu_result = stack_results(cat_cv_result)
cat_gpu_result.groupby(['Metric', 'Dataset']).Value.mean().unstack()

#plot_result(cat_gpu_result, model='CatBoost', fname=f'figures/{algo}_cv_result')
#no access to gpu modified code
plot_result(cat_result, model='CatBoost', fname=f'figures/{algo}_cv_result')
